pkg/operator/sync.go

```diff
@@ -565,6 +630,96 @@ func newTerminationDaemonSet(config *OperatorConfig) *appsv1.DaemonSet {
 	}
 }
 
+func newValidatingWebhook(config *OperatorConfig) *admissionregistrationv1.ValidatingWebhookConfiguration {
+	policy := admissionregistrationv1.Ignore
+	sideEffects := admissionregistrationv1.SideEffectClassNone
+	path := mapiv1.DefaultValidatingHookPath
+	serviceReference := admissionregistrationv1.ServiceReference{
+		Namespace: config.TargetNamespace,
+		Name:      mapiv1.DefaultWebhookName,
+		Path:      &path,
+	}
+	hooks := []admissionregistrationv1.ValidatingWebhook{
+		{
+			AdmissionReviewVersions: []string{"v1"},
+			Name:                    "validation.machine.machine.openshift.io",
+			FailurePolicy:           &policy,
+			SideEffects:             &sideEffects,
+			ClientConfig: admissionregistrationv1.WebhookClientConfig{
+				Service: &serviceReference,
+			},
+			Rules: []admissionregistrationv1.RuleWithOperations{
+				{
+					Rule: admissionregistrationv1.Rule{
+						APIGroups:   []string{machine.GroupName},
+						APIVersions: []string{mapiv1.SchemeGroupVersion.Version},
+						Resources:   []string{"machines"},
+					},
+					Operations: []admissionregistrationv1.OperationType{
+						admissionregistrationv1.Create,
+						admissionregistrationv1.Update,
+					},
+				},
+			},
+		},
+	}
+
+	return &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "machine-api",
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Webhooks: hooks,
+	}
+}
+
+func newMutatingWebhook(config *OperatorConfig) *admissionregistrationv1.MutatingWebhookConfiguration {
+	name := "default.machine.machine.openshift.io"
+	policy := admissionregistrationv1.Ignore
+	path := mapiv1.DefaultMutatingHookPath
+	sideEffects := admissionregistrationv1.SideEffectClassNone
+	serviceReference := admissionregistrationv1.ServiceReference{
```

Comment:
I was under the impression that the CA bundle was a required field and that it had to be set, though it looks as though it wasn't in the YAML definitions, have you verified that this can create the MWC/VWC if they don't exist?

Reply:
It should be fine, as the field is optional:
```golang
	// `caBundle` is a PEM encoded CA bundle which will be used to validate the webhook's server certificate.
	// If unspecified, system trust roots on the apiserver are used.
	// +optional
	CABundle []byte `json:"caBundle,omitempty" protobuf:"bytes,2,opt,name=caBundle"`
```